### DISCUSSION TITLE
Azure Blob Services API - Tests

### DISCUSSION THREAD

**Discussion by GabrielMuzolf** [microsoft/ALAppExtensions#15152] [Q&A] [answered]
Hi,
Currently I am working on module Azure Blob Services API. I would like to add extra methods to get Blob tags in form of XmlDocument and Dictionary of Key:Value pairs via "ABS Client Impl.", now only Set method is available. I wrote a code and now I would like to write some tests. I found out, that Azurity emulator is used for this purpose. I installed it and it works in Microsoft Storage Explorer, so host runs fine. But when i run test functions (for example method PutBlobBlockBlobStreamTest from "ABS Blob Client Test" CU) procedure crash on first request (Create Container) with error "An error occurred while sending the request. Unable to connect to the remote server 6 s 940 ms". Am I doing something incorrectly?
Btw1. Why there is so few tests of this module?
Btw2. Why most of the tests are commented out?
Gabriel

**Comment by SimonOfHH** [comment, marked as answer]
Hi @GabrielMuzolf 👋
Just saw this question. Regarding your Azurite-problem: I assume you develop in a Docker container, correct? And is Azurite also running in the container or on the host? If it's the latter, that would explain why you can't connect from within the container to the host, though.

Regarding "_Why there is so few tests of this module?_": Not enough time I'd say 😅 Feel free to contribute additional tests 🤗
Regarding "_Why most of the tests are commented out?_": I'm not 100% sure, but if I understood it correctly, I think it's because Microsofts internal build-system can't use Azurite yet. And since this repo is basically the same used for the internal builds at Microsoft they deactivated the tests for that. But maybe @mazhelez can give more input here.

If you still have troubles with Azurite/Container setup, please feel free to reach out 😊

**Reply by mazhelez** [reply]
@SimonOfHH's reply was spot on!

**Reply by GabrielMuzolf** [reply]
This is it! Thank you for your answer :)
